content/en/docs/setup/production-environment/dockershim.md

```diff
@@ -0,0 +1,157 @@
+even better fit for your environment.
+
+[CNCF landscape]: https://landscape.cncf.io/category=container-runtime&format=card-mode&grouping=category
+
+
+### What should I look out for when changing CRI implementations?
+
+While the underlying containerization code is the same between Docker and most
+CRIs (including containerd), there are a few differences around the edges. Some
+common things to consider when migrating are:
+- Logging configuration
+- Runtime resource limitations
+- Node provisioning scripts that require Docker or the controlsocket
+- Kubectl plugins that require docker CLI or the control socket
+- Kubernetes tools that require direct access to Docker (e.g. kube-imagepuller)
+- Configuration of functionality like `registry-mirrors` and insecure registries 
+- Other support scripts or daemons that expect docker to be available and are run
+  outside of Kubernetes (e.g. monitoring or security agents)
+
+If you use Kubernetes resource requests/limits or file-based log collection
+DaemonSets then they will continue to work the same, but if you’ve customized
+your dockerd configuration, you’ll need to adapt that for your new container
+runtime where possible.
+
+Another thing to look out for is anything expecting to run [ed: how to say
+docker CLI commands in a way that isn’t repetitive] for system maintenance or
+nested inside a container when building images will no longer work. For the
+former, you can use the [`crictl`][cr] tool as a mostly drop-in replacement
+and for the latter you can use newer container build options like [img], [buildah],
+or [kaniko] that don’t require Docker.
+
+[cr]: https://github.com/kubernetes-sigs/cri-tools
+[img]: https://github.com/genuinetools/img
+[buildah]: https://github.com/containers/buildah
+[kaniko]: https://github.com/GoogleContainerTools/kaniko
+
+For containerd in particular, you can start with their [documentation] to see what
+configuration options are available as you migrate things over.
+
+[documentaiton]: https://github.com/containerd/cri/blob/master/docs/registry.md
+
+For instructions on how to use containerd and CRI-O with Kubernetes, see the
+Kubernetes documentation on [Container Runtimes]
+
+[Container Runtimes]: /docs/setup/production-environment/container-runtimes
+
+
+### Why is dockershim being deprecated?
+
+Maintaining dockershim has become a heavy burden on the Kubernetes maintainers.
+The CRI standard was created to reduce this burden and allow smooth interoperability
+of different container runtimes, however runtimes. Docker itself doesn't currently
+implement CRI, thus the problem. Dockershim was always intended to be a temporary
+solution (hence the name: shim). You can read more about the community discussion
+and planning in the [Dockershim Removal Kubernetes Enhancement Proposal][drkep]
+
+[drkep]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1985-remove-dockershim
+
```

Comment:
Newer features such as cgroups v2 and user namespaces are being implemented in the new CRI runtimes and cutting away from dockershim allows us to move faster in supporting new kernel features for containerization.